src/Mod/PartDesign/Gui/ReferenceSelection.cpp

```diff
@@ -345,12 +347,12 @@ QString getRefStr(const App::DocumentObject* obj, const std::vector<std::string>
         return {};
     }
 
-    if (PartDesign::Feature::isDatum(obj)) {
+    if (PartDesign::Feature::isDatum(obj) || obj->isDerivedFrom(Base::Type::fromName("Sketcher::SketchObject"))) {
         return QString::fromLatin1(obj->getNameInDocument());
     }
-    else if (!sub.empty()) {
+    else if (!sub.empty() && !sub[0].empty()) {

```

Comment:
Can you explain this change? I can't understand why you did that just looking at the code, and usually an empty subname is still a subname so I wonder.